applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/rules.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+'require uci';
+'require form';
+'require shadowsocks-libev as ss';
+
+var conf = 'shadowsocks-libev';
+
+function src_dst_option(s /*, ... */) {
+	var o = s.taboption.apply(s, L.varargs(arguments, 1));
+	o.datatype = 'or(ipaddr,cidr)';
+}
+
+return L.view.extend({
+	load: function() {
+		return uci.load(conf)
+			.then(function() {
+				if (!uci.get_first(conf, 'ss_rules')) {
+					uci.set(conf, uci.add(conf, 'ss_rules', 'ss_rules'), 'disabled', '1');
+				}
+			});
```

Comment:
Is it something wrong? Original code is: https://github.com/openwrt/luci/blob/5c26916d7b31d24268cb1f994f229c08663ad23e/applications/luci-app-shadowsocks-libev/luasrc/model/cbi/shadowsocks-libev/rules.lua#L19-L23

Reply:
How was the code generated?  Possibly the code could work if `uci.add()` returns the newly added section name.  But I do not like it writing this way.  It's not bad for reading and hard to debug if things went wrong in that line.

Reply:
`uci.add()` returns the section name. I tested and it works, just want to know if there's better way doing this.